Training/1.0 Introduction.md

```diff
@@ -11,38 +11,83 @@ First we we learn a little bit about coding in Python, although many of you will
 
 List all the Data Types of Python with examples.  
 
-```
-
-```
+``` str = 'word'
+``` int = 2
+``` float = 2.1
+``` complex = 2j
+``` list = ["hello", "goodbye"]
+``` tuple = ("hello", "goodbye")
+``` range = range(2)
+``` dict = {"eat":"chips", "drink":"water"}
+``` set = {"hello", "goodbye"}
+``` frozenset = ({"hello", "goodbye"})
+``` bool = True
+``` bytes = b"Cheers"
+``` bytearray = bytearray(2)
+``` memoryview = memoryview(bytes(2))
+``` NoneType = None
 
 ### Task 2: 
 
 Other useful ways of storing data are in List, Arrays and Tuples. Create one of each and assign them to the variables a, b and c respectively. 
 
-```
-
+``` listx = ["a", "b", "c"]
+``` arrayx = listx
+``` tuplex =("a", "b", "c")
 ```
 
 ### Task 3: 
 
 We are also interested in becoming good software developers so we will need to use conditional, loops. Write an statement where the summation of a variable x+1 will be calculated if the value of x is greater than 2. 
 
-```
+``` x = input("please input a number")
+``` x = float(x)
+``` if 2<<(x+1):
+```     y = x+1
+``` else:
+        continue
+    print(y)
 
 ```
 
 Now incorporate a for loop to calculate the value of n(x+1) for n repetitions where n=3. Store each value of the for loop in an array named y.  
 
-```
+``` n = 0
+``` y = []
+``` x = input("please input a number")
+``` x = float(x)
+``` for num in n:
+```     if n <= 3:
+```         z = n(x+1)
+```         y.append(z)
+```         n = n+1
+```         continue
+```     else:
+```         break
+``` print(y)
+        
```

Comment:
Good attempt, but there's several issues here:
1) To create and append to an array, you need to use numpy.
2) In a for loop, the iteration sequence needs to either be a list (i.e. `n = [1,2,3]` `for num in n:`) or a range (i.e. `for num in range(3):`. 
The approach you used with n as an incrementing counter is commonly used for while loops, not for loops.
3) If you defined the iteration sequence correctly, you wouldn't need the if statement inside the for loop.

Correct solution:
`import numpy as np`
`y=np.array([])`
`n=3`
`x=2`
`for i in range(n):`
`    prd=n*(x+1)`
`    y=np.append(y,prd)`